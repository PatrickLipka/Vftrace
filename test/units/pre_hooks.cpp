#include <iostream>
using namespace std;

int before_main(void) {
  cout<<"before main\n";
  return 137;
}

static int dummy = before_main(); 

int main() {
  cout<<"inside main\n";
  cout<<dummy<<"\n";
  return 0;
}
