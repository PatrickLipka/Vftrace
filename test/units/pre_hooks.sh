#!/bin/bash

set -x

test_name=pre_hooks
logfile=${test_name}_all.log
output_file=${test_name}.out
ref_file=${srcdir}/ref_output/${test_name}.out

rm -f ${output_file}

if [ "x${HAS_MPI}" == "xYES" ]; then
   ${MPI_EXEC} ${MPI_OPTS} ${NP} 1 ./${test_name} > ${output_file} || exit 1
else
   ./${test_name} > ${output_file} || exit 1
fi

diff ${output_file} ${ref_file} || exit 1

cat ${logfile}
# Count how often the before_main function appears in the profile
count=$(cat ${logfile} | grep "before_main")
if [[ "${count}" -ne "0" ]] ; then
   echo "Expected not to find \"before_main\" in logfile, but found it ${count} times."
   exit 1;
fi
