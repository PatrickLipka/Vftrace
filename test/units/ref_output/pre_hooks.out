before main
inside main
137
